### PR TITLE
Multi bucket gcs handler

### LINKED ISSUE #9994: [Enhancement] Add Multi-bucket Support for the Google Cloud Storage Integration
### Short description and motivation for the proposed feature

The goal of this issue is to support multi-bucket support via the [Google Cloud Storage integration](https://github.com/mindsdb/mindsdb/tree/main/mindsdb/integrations/handlers/gcs_handler). At the moment, this integration requires a bucket to be passed in as a connection argument and only the objects in that bucket can be queried. However, it would be useful to allow users to query data across buckets in their Google Cloud account.

[Here](https://github.com/mindsdb/mindsdb/pull/9879) is a good example of how this has been achieved for the Amazon S3 integration.

## Resources :bulb: 

:checkered_flag: To get started, please refer to the following resources:

* :books:  [Building Application Handler docs](https://docs.mindsdb.com/contribute/app-handlers): This documentation will guide you through the process of building a custom handler for MindsDB. It contains essential information on the handler structure, methods, and best practices. 
*  :point_right: [Google Cloud Storage API Handler](https://github.com/mindsdb/mindsdb/tree/main/mindsdb/integrations/handlers/gcs_handler) and  [S3 API Handler](https://github.com/mindsdb/mindsdb/tree/main/mindsdb/integrations/handlers/s3_handler)

## Next Steps :male_detective: :female_detective: 
 
-   Familiarize yourself with Google Cloud Storage and the existing integration for it.
-   Familiarize yourself with how this enhancement has been achieved for the Amazon S3 integration by studying the PR linked above.
-   Implement the feature.
-   Thoroughly test the integration to ensure proper functionality.
-   Submit a pull request with the support for querying across buckets.

## :loudspeaker: Additional rewards :trophy:

Each enchamcement brings 25 points for SWAG and entry into the draw for a  :point_down: :
- :computer:  [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
- :shirt: :bear:  [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)
- :incoming_envelope:  For more info check out https://mindsdb.com/hacktoberfest/ :point_left: 

### Video or screenshots

_No response_

### Describe some possible solutions

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Better-Boy**
## Description

Please include a summary of the change and the issue it solves. 

Fixes #9994 

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [ ] I have attached a brief loom video or screenshots showcasing the new functionality or change.

## Checklist:

- [ ] My code follows the style guidelines(PEP 8) of MindsDB.
- [ ] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/gcs_handler/gcs_tables.py [hunk 1]
```diff
@@ -9,14 +9,25 @@ class ListFilesTable(APIResource):
...
              targets: List[str] = None,
              conditions: List[FilterCondition] = None,
+             limit: int = None,
              *args, **kwargs) -> pd.DataFrame:
 
-        tables = self.handler._get_tables()
+        buckets = None
+        for condition in conditions:
+            if condition.column == 'bucket':
+                if condition.op == FilterOperator.IN:
```
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/gcs_handler/gcs_tables.py:18]: `FilterOperator` has been used, but not imported.

File: mindsdb/integrations/handlers/gcs_handler/README.md [hunk 2]
```diff
@@ -19,17 +19,20 @@ CREATE DATABASE gcs_datasource
 WITH
```
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/gcs_handler/README.md:19]: Can you please update the documentation based on  the changes made here?
https://github.com/mindsdb/mindsdb/pull/9979

**Review by MinuraPunchihewa — CHANGES_REQUESTED** [review]
Can you please test this thoroughly and post some screenshots after the few changes I've mentioned below?

File: mindsdb/integrations/handlers/gcs_handler/gcs_tables.py (hunk 1 above)
Comment (Better-Boy) [reply mindsdb/integrations/handlers/gcs_handler/gcs_tables.py:18]: Imported

File: mindsdb/integrations/handlers/gcs_handler/README.md (hunk 2 above)
Comment (Better-Boy) [reply mindsdb/integrations/handlers/gcs_handler/README.md:19]: There's no markdown files for gcs in the docs folder. Shall I create one?

Comment (MinuraPunchihewa) [reply mindsdb/integrations/handlers/gcs_handler/README.md:19]: We can tackle that later. For now, please post screenshots of this feature working.

Comment (Better-Boy) [reply mindsdb/integrations/handlers/gcs_handler/README.md:19]: It's working fine.

Comment (MinuraPunchihewa) [reply mindsdb/integrations/handlers/gcs_handler/README.md:19]: Please also post a screenshot of querying a file in a bucket providing the full path.
E.g., `my-bucket/my-dir/my-file.csv`

**Review by MinuraPunchihewa — COMMENTED** [review]
@Better-Boy Please solve the lint issues to fix the failing CI and provide the screenshot I have requested.

File: mindsdb/integrations/handlers/gcs_handler/README.md (hunk 2 above)
Comment (Better-Boy) [reply mindsdb/integrations/handlers/gcs_handler/README.md:19]: Done.  Tested.

**Comment by Better-Boy** [comment]
> @Better-Boy Please solve the lint issues to fix the failing CI and provide the screenshot I have requested.

Done. Fixed.

**Review by MinuraPunchihewa — APPROVED** [review]

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**